groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -704,55 +1010,415 @@ struct Optional_Data<TYPE, true> : public Optional_DataImp<TYPE> {
+    BSLMF_NESTED_TRAIT_DECLARATION_IF(optional,
+                                      bsl::is_trivially_copyable,
+                                      bsl::is_trivially_copyable<TYPE>::value);
+    // Workaround for C++03 'bsl::is_trivially_copyable' trait.
+#endif  //BSLS_LIBRARYFEATURES_HAS_CPP11_BASELINE_LIBRARY
+
+    // CREATORS
+    optional();
+        // Create a disengaged 'optional' object.  Use the currently installed
+        // default allocator to supply memory for future 'value_type' objects.
+
+    optional(bsl::nullopt_t);                                       // IMPLICIT
+        // Create a disengaged 'optional' object.  Use the currently installed
+        // default allocator to supply memory for future 'value_type' objects.
+
+    optional(const optional& rhs);                                  // IMPLICIT
+        // If 'rhs' contains a value, initialize the contained value using
+        // '*rhs'. Otherwise, create a disengaged 'optional'.  Use the
+        // currently installed default allocator to supply memory for this and
+        // any future 'value_type' objects.
+
+    optional(BloombergLP::bslmf::MovableRef<optional> rhs);         // IMPLICIT
+        // If 'rhs' contains a value, initialize the contained 'value_type'
+        // object by move construction from '*rhs'. Otherwise, create a
+        // disengaged 'optional'.  Use the allocator from rhs to supply memory
+        // for this and any future 'value_type' objects.  'rhs' is left in a
+        // valid, but unspecified state.
+
+    // Because there are no default arguments in C++03, the case of
+    // 'ANYTYPE==TYPE' is written out separately.
+    template <class ANY_TYPE>
+    optional(BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE) rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_SAME(TYPE, ANY_TYPE)
+        BSLSTL_OPTIONAL_ENABLE_IF_NOT_EXPLICIT_CONSTRUCT(TYPE, ANY_TYPE))
+        // Create an 'optional' object having the value of the specified 'rhs'
+        // object.  Use the currently installed default allocator to supply
+        // memory for future 'value_type' objects.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        this->emplace(BSLS_COMPILERFEATURES_FORWARD(ANY_TYPE, rhs));
+    }
+
+    template <class ANY_TYPE>
+    explicit optional(BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE) rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_SAME(TYPE, ANY_TYPE)
+        BSLSTL_OPTIONAL_ENABLE_IF_EXPLICIT_CONSTRUCT(TYPE, ANY_TYPE))
+        // Create an 'optional' object having the value of the specified 'rhs'
+        // object.  Use the currently installed default allocator to supply
+        // memory for future 'value_type' objects.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        this->emplace(BSLS_COMPILERFEATURES_FORWARD(ANY_TYPE, rhs));
+    }
```

Comment:
What's the purpose of having two constructors with opposite conditions for enable_if having exactly the same imp?

Also, we don't care about VS2010 anymore.  Do you have the ability to check that having out-of-line definitions here would work this with VS2013?

Reply:
one is explicit, one isn't. 
I'll check if out-of-line definitions work with VS2013

Reply:
Yep, that's what I missed, thank you!

Reply:
Most of the functions were fine out of class in VS2013. However, three were causing an odd compilation issue I could not explain, so I moved them back in.